docs/proposal-platforms.md

```diff
@@ -235,31 +235,183 @@ suite-run functionality:
-       (defaults to the platform name).
-     * If multiple platforms share the same root filesystem then use a common
-       id.
-     * Would need to consider the case where the root filesystem is shared but
-       log, work or share differ.
-
-2. Support for moving the share, work & log directories to different locations
-   with optional symlinks to the root directory. Similarly support for moving
-   the root directory with symlink to the `cylc-run` directory.
+   * We will modify this to work on an install target basis rather than a
+     platform basis.
+     * Configure the id of the target used on a particular platform
+       (`install target`: defaults to the platform name).
+     * If multiple platforms share the same filesystems then use a common
+       `install target`.
+     * Note that, when we add we support for separate client keys, these will
+       now be per `install target` rather than per platform.
+     * Cylc 7 creates a `.service/uuid` file to detect whether a remote host
+       shares the same filesystem as the scheduler host - we can remove this
+       functionality. Instead we should check for the existence of a client key
+       and fail the remote-init if one is found for a different `install target`
+       (different targets used within the same workflow should not share the
+       same run directory).
+   * Currently, the files which need to be installed are transferred as a tar
+     file via stdin. This has the advantage that the remote init can be done as
+     a single ssh. We will change the installation to use rsync which will
+     require a second ssh but this will be better for installing more files and
+     allow the logging to report which files have been changed after a reload.
+     * If we change cylc to re-use ssh connections by default (using
+       `ControlPersist`) then this will avoid the ssh overhead (and will
+       benefit other ssh connections as well).
+     * We will probably need a third ssh connection for transferring the client
+       public key.
+   * The remote installation will include particular files/directories.
+     * Items to include:
+       `.service/server.key .service/contact app/ bin/ etc/ lib/`.
+     * We will use the `--delete` rsync option so that any files removed from
+       from installed directories also get removed from the install targets on
+       reload/restart.
+     * Does this need to be configurable? (i.e. configure additional items to
+       include)? If so, global and/or workflow level?
+       * Is there any reason why workflow designers can't just stick to the list
+         of supported directories for files that need to be installed?
+       * Are there many existing Rose suites which rely on other directories
+         being installed? A quick search implies there are quite a few suites
+         using a wide range of sub-directories such as `python`, `util`, `data`
```

Comment:
Are these reasonable? I can't help wondering if `python` and `util` should be under `lib` and `bin` respectively. `data` could feasibly go under `etc`, depending on how it is used. Can we sample some of these cases and see if it's a suite design fail or a genuine need?

Reply:
I agree these shouldn't be needed (see previous bullet). However, these suites already exist and currently work with rose suite-run. Hence my proposal which allows affected users to modify their global config so that they can still run these suites (until the suites can be modified to use the supported directories).